### PR TITLE
Live 61 add routing for hls endpoint

### PR DESCRIPTION
**Pull request by Karolk99**

### CODE REVIEW

File: assets/js/pages/Event.tsx [hunk 1]
```diff
@@ -45,16 +37,47 @@ const initEventInfo = () => {
   };
 };
 
+const getEventInfo = (
+  eventInfo: EventInfo,
+  setEventInfo: React.Dispatch<React.SetStateAction<EventInfo>>
+) => {
+  const csrfToken = document.querySelector("meta[name='csrf-token']")?.getAttribute("content");
+
+  fetch("http://localhost:4000/webinars/" + eventInfo.link, {
```
Comment (Rados13) [inline assets/js/pages/Event.tsx:46]: As above mentioned use something like `window.origin` or `window.location.href`.

File: assets/js/pages/Event.tsx [hunk 2]
```diff
@@ -63,72 +86,56 @@ const Event = () => {
...
-    if (privateChannel) {
-      createPrivateChannel(
-        privateChannel,
-        eventChannel,
-        eventInfo.username,
-        setPresenterPopupState
-      );
+    const alreadyJoined = privateChannel?.state === "joined";
+    if (name && !alreadyJoined){
+        const channel = socket.channel("private:" + eventInfo.link + ":" + name, {});
```
Comment (Rados13) [inline assets/js/pages/Event.tsx]: Use template literals.

File: assets/js/utils/rtcUtils.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+  webrtc.join({
+    displayName: name,
+  });
+
+  webrtcChannel.on("mediaEvent", (event) => webrtc.receiveMediaEvent(event.data));
+
+  webrtcChannel.on("playlistPlayable", ({ playlistId }) => {
+    // setPlayerInfo(playlistId);
+    alert("playlistPlayable");
```
Comment (Rados13) [inline assets/js/utils/rtcUtils.ts:77]: I would add a comment that this should be propagated in the future.

File: lib/membrane_live/event.ex [hunk 4]
```diff
@@ -170,8 +179,7 @@ defmodule MembraneLive.Event do
     Membrane.Logger.error("Endpoint #{inspect(endpoint_id)} has crashed!")
     peer_channel = state.peer_channels[endpoint_id]
 
-    # TODO: handle HLS endpoint error.
-    error_message = "Endpoint has crashed."
+    error_message = "WebRTC endpoint has crashed, please refresh the page to reconnect"
```
Comment (Rados13) [inline lib/membrane_live/event.ex:182]: Also, the HLS endpoint can crush.

File: lib/membrane_live/event.ex [hunk 5]
```diff
@@ -206,31 +214,31 @@ defmodule MembraneLive.Event do
...
+    state = put_in(state, [:playlist_idl], playlist_idl)
+    MembraneLiveWeb.Endpoint.broadcast!("event:" <> state.event_id, "playlist_playable", %{playlist_idl: playlist_idl})
     {:noreply, state}
   end
 
   @impl true
-  def handle_info({:cleanup, _clean_function, stream_id}, state) do
-    StorageCleanup.remove_directory(stream_id)
+  def handle_info({:cleanup, _clean_function, _stream_id}, state) do
+    # StorageCleanup.remove_directory(stream_id)
```
Comment (Rados13) [inline lib/membrane_live/event.ex]: If we are not using this dependency remove it.

**Review by Rados13 — CHANGES_REQUESTED** [review]

File: assets/js/components/PresenterStreamArea.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+  useEffect(() => {
+    // TODO: self-view
+    // implement changing audio/video source
+    if (!isConnected && presenters.includes(username)) {
+      connectWebrtc(eventChannel, username, streamsAvailable, setStreamsAvailable);
+      isConnected = true;
+    } else if (isConnected && !presenters.includes(username)) {
+      // TODO: leave WebRTC connection
+    }
+  }, [presenters]);
```
[the remark this answers is not in the document]
Comment (Karolk99) [reply assets/js/components/PresenterStreamArea.tsx:26]: yes

File: assets/js/components/RtcPlayer.tsx [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+const RtcPlayer = ({ name, presenterStreams, streamsAvailable }: RtcPlayerProps) => {
+  const videoRef = useRef<HTMLVideoElement>(null);
+  const audioRef = useRef<HTMLAudioElement>(null);
+
+  useEffect(() => {
+    if (
+      name in presenterStreams &&
+      streamsAvailable[name] &&
+      videoRef.current != null &&
+      audioRef.current != null
```
[the remark this answers is not in the document]
Comment (Karolk99) [reply assets/js/components/RtcPlayer.tsx:18]: will be done in LIVE-41

**Review by Rados13 — APPROVED** [review]
Only little changes left

**Pull request merged**